lib/chef/provider/user/smartos.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Author:: Dave Eddy (<dave@daveeddy.com>)
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'chef/mixin/shell_out'
+require 'chef/provider/user/useradd'
+
+class Chef
+  class Provider
+    class User
+      class SmartOS < Chef::Provider::User::Useradd
+        provides :user, os: 'solaris2', platform_family: 'smartos'
+
+        def check_lock
+          shadow_line = shell_out!('getent', 'shadow', new_resource.username).stdout.strip
```

Comment:
also, this line will exit if the user does not exist... it can be modified if needed to return `nil` in that case if needed.  I figured it is an error to call `check_lock` on a user that does not yet exist.